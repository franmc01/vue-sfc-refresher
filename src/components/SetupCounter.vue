<script lang="ts">
import { defineComponent } from 'vue'
import useCounter from '@/composables/useCounter'

export default defineComponent({
  props: {
    initialValue: {
      type: Number,
      required: false,
      default: 0
    }
  },
  setup(props){
    // const counter = ref(props.initialValue)
    // const squareCounter = computed(() => counter.value * 2)

    // const increment = () => counter.value++
    // const decrement = () => counter.value--

    const { counter, squareCounter, increment, decrement } = useCounter(props.initialValue)
    return {
      // squareCounter: computed(() => counter.value * 2),
      squareCounter,
      counter,
      increment,
      decrement
    }
  }
})

</script>

<template>
  <section>
    <h2 data-test="initial-value">Setup Counter: {{counter}}</h2>
    <h3 data-test="square-value">Square counter: {{squareCounter}}</h3>
    <div>
      <button id="increment" @click="increment">+1</button>
      <button id="decrement" @click="decrement">-1</button>
    </div>
  </section>
</template>

<style scoped>

</style>
