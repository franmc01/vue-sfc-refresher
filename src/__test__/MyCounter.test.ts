import { mount } from '@vue/test-utils'
import SetupCounter from '@/components/SetupCounter.vue'
import { expect } from 'vitest'

describe('SetupCounter', () => {
  test('Render component with default value 0', async () => {
    const wrapper = mount(SetupCounter);
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Render component with 6 when increment', async () => {
    const wrapper = mount(SetupCounter, {
      props: {
        initialValue: 5,
      }
    });
    await wrapper.findAll('button')[0].trigger('click')
    expect(wrapper.find('h2').text()).toContain('6')
  })

  test('Render component with 5 when decrement', async () => {
    const wrapper = mount(SetupCounter, {
      props: {
        initialValue: 7,
      }
    });

    await wrapper.findAll('button')[1].trigger('click')
    await wrapper.findAll('button')[1].trigger('click')

    expect(wrapper.find('h2').text()).toContain('5')
  })

  test('Render component with initial value 7 and square 14', () => {
    const wrapper = mount(SetupCounter, {
      props: {
        initialValue: 7,
      }
    });
    expect(wrapper.find('[data-test="initial-value"]').text()).toContain('7')
    expect(wrapper.find('h3').text()).toContain('14')
  })
})
