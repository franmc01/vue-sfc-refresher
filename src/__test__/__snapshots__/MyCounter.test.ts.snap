// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SetupCounter > Render component with default value 0 1`] = `
"<section>
  <h2 data-test="initial-value">Setup Counter: 0</h2>
  <h3 data-test="square-value">Square counter: 0</h3>
  <div><button id="increment">+1</button><button id="decrement">-1</button></div>
</section>"
`;
